vcpkg build failure

Bootstrap build fails on Win10 /w vs2015

Environment:
- OS: Windows 10
- Compiler: vs2015 14.0.25431.01 Update 3

Steps to reproduce the behavior:
1. git clone vkpkg
2. bootstrap-vcpkg.bat

I'm expecting it to build using vs2015. But I get this:

..\src\vcpkg\install.cpp(845): error C2280: 'vcpkg::PortFileProvider::BaselineProvider::BaselineProvider(const vcpkg::PortFileProvider::BaselineProvider &)': attempting to reference a deleted function [E:\vcpkg\toolsrc\windows-bootstrap\vcpkg.vcxproj]
  E:\vcpkg\toolsrc\include\vcpkg/portfileprovider.h(84): note: compiler has generated 'vcpkg::PortFileProvider::BaselineProvider::BaselineProvider' here

Looking at the error I can't even figure out why it's trying to call the default constructor there... both calls in the lambda to construct the BaselineProvider are using non-default parameters. I did spend a little time just trying to fix the compile error but I couldn't figure it out. Seems like the compiler is just wrong here...

Hi, I got the same issue after pulling the repo. Previously I was at 539a182d3, and everything was fine, I pulled 0a457099416a843f442b6c2275b2ec57239b50d9 and I have this issue. 

On Windows 10, VS2015


`Building vcpkg.exe ...

  install.cpp
..\src\vcpkg\install.cpp(845): error C2280: 'vcpkg::PortFileProvider::BaselineProvider::BaselineProvider(const vcpkg::PortFileProvider::BaselineProvider &)': attempting to reference a deleted function [C:\tools\vcpkg\toolsrc\windows-bootstrap\vcpkg.vcxproj]
  C:\tools\vcpkg\toolsrc\include\vcpkg/portfileprovider.h(84): note: compiler has generated 'vcpkg::PortFileProvider::BaselineProvider::BaselineProvider' here
C:\tools\vcpkg\scripts\bootstrap.ps1 : Building vcpkg.exe failed. Please ensure you have installed Visual Studio with the Desktop C++ workload and the Windows SDK for Desktop C++.
Au caractère Ligne:1 : 4
+ & {& 'C:\tools\vcpkg\scripts\bootstrap.ps1' }
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,bootstrap.ps1

ScriptHalted
Au caractère C:\tools\vcpkg\scripts\bootstrap.ps1:411 : 5
+     throw
+     ~~~~~
    + CategoryInfo          : OperationStopped: (:) [], RuntimeException
    + FullyQualifiedErrorId : ScriptHalted
`

--- a/toolsrc/src/vcpkg/install.cpp
+++ b/toolsrc/src/vcpkg/install.cpp
@@ -839,20 +839,20 @@ namespace vcpkg::Install
             if (args.versions_enabled())
             {
                 PortFileProvider::VersionedPortfileProvider verprovider(paths);
-                auto baseprovider = [&]() -> PortFileProvider::BaselineProvider {
+                auto baseprovider = [&]() -> std::unique_ptr<PortFileProvider::BaselineProvider> {
                     if (auto p_baseline = manifest_scf.core_paragraph->extra_info.get("$x-default-baseline"))
                     {
-                        return PortFileProvider::BaselineProvider(paths, p_baseline->string().to_string());
+                        return std::make_unique<PortFileProvider::BaselineProvider>(paths, p_baseline->string().to_string());
                     }
                     else
                     {
-                        return PortFileProvider::BaselineProvider(paths);
+                        return std::make_unique<PortFileProvider::BaselineProvider>(paths);
                     }
                 }();
 
                 auto install_plan =
                     Dependencies::create_versioned_install_plan(verprovider,
-                                                                baseprovider,
+                                                                *baseprovider,
                                                                 var_provider,
                                                                 manifest_scf.core_paragraph->dependencies,
                                                                 manifest_scf.core_paragraph->overrides,

--- a/toolsrc/src/vcpkg/install.cpp
+++ b/toolsrc/src/vcpkg/install.cpp
@@ -842,7 +842,8 @@ namespace vcpkg::Install
                 auto baseprovider = [&]() -> std::unique_ptr<PortFileProvider::BaselineProvider> {
                     if (auto p_baseline = manifest_scf.core_paragraph->extra_info.get("$x-default-baseline"))
                     {
-                        return std::make_unique<PortFileProvider::BaselineProvider>(paths, p_baseline->string().to_string());
+                        return std::make_unique<PortFileProvider::BaselineProvider>(paths,
+                                                                                    p_baseline->string().to_string());
                     }
                     else
                     {

